### ISSUE TITLE
Export all pools

### ISSUE THREAD

**Issue by FransUrbo** [zfsonlinux/zfs#3203]
A command similar to <code>zpool import -a</code> would be nice:

    zpool export -a

instead of having to loop through all individually.

This doesn't sound to difficult and I would be prepared to do the job, if someone could point me in the right direction.

**Comment by ilovezfs** [comment]
Yes, please. I am so tired of my zexport_all script:
```
zpool list -H | awk -F '\t' '{print $1;}' | xargs -I % zpool export %
```

**Comment by behlendorf** [comment]
I actually found myself wanting the exact same thing the other day.

@FransUrbo the place to start here is in `zpool_do_export()`.  The code will already loop over all the pools specified on the command line so it should just be a matter of adding the `-a` option to generate the list of pools and then improving the error handling.  Take a look at `zpool_list_get` to see how to iterate over all the known pools to get their names.

For example, this works today, `zpool export tank1 tank2 tank3`.

**Comment by ilovezfs** [comment]
@behlendorf Which behavior do you think will be best in the case of a system on ZFS root?
1. Automatically skip the root pool
2. Fail immediately if -a is specified
3. Let the command fail naturally when it discovers that pool is busy

Also, in general, if along the way one pool fails to export, should the iteration continue or stop? If it is allowed to continue trying to export the remainder of the pools, how should it affect the return code that one (or more) failed to export?

**Comment by behlendorf** [comment]
Closing.  This was implemented and is in 0.6.5.
